pkg/sfu/connectionquality/connectionstats.go

```diff
@@ -181,3 +215,60 @@ func ToAnalyticsVideoLayer(layer int, layerStats *buffer.LayerStats) *livekit.An
 		Frames:  layerStats.Frames,
 	}
 }
+
+func (cs *ConnectionStats) getBytesFramesFromStreams(streams []*livekit.AnalyticsStream) (totalBytes uint64, totalFrames uint32, maxLayer int32) {
+	type stat struct {
+		TotalBytes  uint64
+		TotalFrames uint32
+	}
+	layerStats := make(map[int32]stat)
+
+	maxLayer = -1
+	hasLayers := false
+	for _, stream := range streams {
+		// get frames/bytes/packets from video layers if available. Store per layer in layerStats map
+		if len(stream.VideoLayers) > 0 {
+			hasLayers = true
+			// find max quality 0(LOW), 1(MED), 2(HIGH), 3(OFF)
+			videoQuality := int32(-1)
+			for _, layer := range stream.VideoLayers {
+				if stats, ok := layerStats[layer.Layer]; ok {
+					stats.TotalFrames += layer.GetFrames()
+					stats.TotalBytes += layer.GetBytes()
+					layerStats[layer.Layer] = stats
+				} else {
+					newStats := stat{
+						TotalBytes:  layer.GetBytes(),
+						TotalFrames: layer.GetFrames(),
+					}
+					layerStats[layer.Layer] = newStats
+				}
+
+				// if layer is off or of lower quality than processed, skip updating maxLayer
+				if (layer.Layer == int32(livekit.VideoQuality_OFF)) || (videoQuality > layer.Layer) {
```

Comment:
We should not compare this. `layer.Layer` should not be `OFF`. We use numbers 0, 1, 2 for layers in `sfu` pkg. That does not map 1:1 to the protocol defined qualities.